src/backend/utils/adt/interpolate.c

```diff
@@ -300,7 +300,7 @@ linterp_int64(PG_FUNCTION_ARGS)
 	else 
 	{
 		r = round(y0+p*(y1-y0));
-		if ( r < LONG_MIN || r > LONG_MAX )
+		if ( r < (float8)LONG_MIN || r > (float8)LONG_MAX )
```

Comment:
gcc 10 does not issues an error without this change

Reply:
GCC in docker:
```
interpolate.c:303:28: warning: implicit conversion from 'long' to 'float8' (aka 'double') changes value from 9223372036854775807 to 9223372036854775808 [-Wimplicit-const-int-float-conversion]
  303 |                 if ( r < LONG_MIN || r > LONG_MAX )
      |                                        ~ ^~~~~~~~
/usr/bin/../lib/clang/17/include/limits.h:47:19: note: expanded from macro 'LONG_MAX'
   47 | #define LONG_MAX  __LONG_MAX__
      |                   ^~~~~~~~~~~~
<built-in>:56:22: note: expanded from macro '__LONG_MAX__'
   56 | #define __LONG_MAX__ 9223372036854775807L
      |                      ^~~~~~~~~~~~~~~~~~~~
1 warning generated.
```